### ISSUE TITLE
updating `$TEALDEER_CACHE_DIR` not recognized on macos

### ISSUE THREAD

**Issue by shamindras** [dbrgn/tealdeer#312]
Thank you for creating this great package.

I'm on macos and using the `XDG` specification. So I have the following paths setup in my `.zshrc`

``` bash
export XDG_CONFIG_HOME="${HOME}/.config"
export XDG_CACHE_HOME="${HOME}/.cache"
```

I then installed the latest `tealdeer` (`v 1.6.1`) using `brew`, and setup the following paths in my `.zshrc`

``` bash
export TEALDEER_CACHE_DIR="${TEALDEER_CACHE_DIR:-$XDG_CACHE_HOME/tealdeer}"
export TEALDEER_CONFIG_DIR="${TEALDEER_CONFIG_DIR:-$XDG_CONFIG_HOME/tealdeer}"
```

I got the (expected) deprecation warning on `TEALDEER_CACHE_DIR` as follows:

``` bash
> tldr --show-paths
Warning: The $TEALDEER_CACHE_DIR env variable is deprecated, use the `cache_dir` option in the config file instead.
Config dir:       /Users/shamindras/.config/tealdeer/ (env variable)
Config path:      /Users/shamindras/.config/tealdeer/config.toml
Cache dir:        /Users/shamindras/.cache/tealdeer (env variable)
Pages dir:        /Users/shamindras/.cache/tealdeer/tldr-pages/
Custom pages dir: /Users/shamindras/Library/Application Support/tealdeer/pages/ (OS convention)
```

I then commented out the line in my `.zshrc`

``` bash
# export TEALDEER_CACHE_DIR="${TEALDEER_CACHE_DIR:-$XDG_CACHE_HOME/tealdeer}"
```

and added the following lines in my `/Users/shamindras/.config/tealdeer/config.toml` config above as per the [recommendation](https://dbrgn.github.io/tealdeer/config_directories.html#cache_dir):

``` toml
[directories]
cache_dir = "/Users/shamindras/.cache/tealdeer" # need to hardcode path
```

But when I restart alacritty, and thus `zsh` (and even my computer), I noticed that the `cache_dir` is not recognized:

```
> tldr --show-paths
Config dir:       /Users/shamindras/.config/tealdeer/ (env variable)
Config path:      /Users/shamindras/.config/tealdeer/config.toml
Cache dir:        /Users/shamindras/Library/Caches/tealdeer (OS convention)
Pages dir:        /Users/shamindras/Library/Caches/tealdeer/tldr-pages/
Custom pages dir: /Users/shamindras/Library/Application Support/tealdeer/pages/ (OS convention)
```

The deprecation warning for `TEALDEER_CACHE_DIR` is now gone, but the macOS convention cache directory is now used. 

How do I fix this to point to my specified `cache_dir` above? Any assistance greatly appreciated.

**Comment by dbrgn** [comment]
This is probably a duplicate of #311, right?

**Comment by dbrgn** [comment]
Is the config file working otherwise (e.g. if you change colors)?

**Comment by shamindras** [comment]
> Is the config file working otherwise (e.g. if you change colors)?

Great suggestion. I now debugged the issue thanks to this. Basically my `Config path:` symlink was broken.
So the file 

``` bash
Config path:      /Users/shamindras/.config/tealdeer/config.toml
```

didn't exist! I simply ensured that this was linked correctly, and it worked.

Your suggestion to try changing the config helped, since I set all display colors to `"red"`, and that didn't work. So I re-checked the existence of the `config.toml` file, which showed the symlink issue.

Sorry for the issue on my end, and thanks for your suggestion and this great tool. Feel free to close.
